packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -185,28 +176,26 @@ async function executePreview(
     return;
   }
 
-  if (platformSelection.id === PreviewPlatformType.Desktop) {
-    await startServer(true, componentName, startTime);
-    return;
-  }
-
   // 2. Prompt user to select a target device
-  let targetDevice: string;
-  try {
-    const targetName = await selectTargetDevice(platformSelection);
-    if (targetName === undefined) {
-      vscode.window.showInformationMessage(commandCancelledMessage);
+  let targetDevice: string | undefined;
+  if (platformSelection.id !== PreviewPlatformType.Desktop) {
+    try {
+      const targetName = await selectTargetDevice(platformSelection);
+      if (targetName === undefined) {
+        vscode.window.showInformationMessage(commandCancelledMessage);
+        return;
+      } else {
+        targetDevice = targetName;
+      }
+    } catch {
+      // exception has already been logged
       return;
-    } else {
-      targetDevice = targetName;
     }
-  } catch {
-    // exception has already been logged
-    return;
   }
 
   // 3. Determine project root directory and path to the config file
   const projectRootDir = getProjectRootDirectory(resourcePath);
+  const componentPath = getRelativeComponentPath(resourcePath, projectRootDir);
```

Comment:
unlike the local dev server that uses a component name with the syntax `c/name`, LWR server needs the path to a component (relative to its project root)